### DISCUSSION TITLE
Dashboard WYSIWYG (q-editor) Waiting For LaTEX

### DISCUSSION THREAD

**Discussion by stepheweffie** [zauberzeug/nicegui#2258] [Show and tell]
I'll be adding some bells this weekend. 
<img width="1440" alt="Screen Shot 2023-12-29 at 4 47 51 AM" src="https://github.com/zauberzeug/nicegui/assets/1074503/874f014e-c179-4d72-93b1-a0e6466e9e05">
<img width="1440" alt="Screen Shot 2023-12-29 at 4 48 09 AM" src="https://github.com/zauberzeug/nicegui/assets/1074503/5db39ff9-d556-4b9d-81b6-e555ae8584b5">

**Comment by stepheweffie** [comment]
```
JavaScript did not respond in time
Traceback (most recent call last):
  File "/Users/savantlab/PycharmProjects/nice-gui-blog/env/lib/python3.8/site-packages/nicegui/events.py", line 407, in wait_for_result
    await result
  File "/Users/savantlab/PycharmProjects/nice-gui-blog/blog/dash.py", line 308, in post_create_modal
    await ui.run_javascript(vue_js_script)
  File "/Users/savantlab/PycharmProjects/nice-gui-blog/env/lib/python3.8/site-packages/nicegui/client.py", line 184, in send_and_wait
    raise TimeoutError('JavaScript did not respond in time')
TimeoutError: JavaScript did not respond in time
```
```

with ui.expansion("WSIWYG Editor", icon='editor').classes('w-full ml-5'):
    ui.html(vue_html_content)
    await ui.run_javascript(vue_js_script)
```

I have the kitchen sink q-editor and script in doc strings. It functions even with the above error.

**Reply by rodja** [reply]
If your `vue_js_script` does not return anything, you should remove the `await`.

**Reply by stepheweffie** [reply]
Why would I not await it?

```
raise RuntimeError('The current slot cannot be determined because the slot stack for this task is empty.\n'
RuntimeError: The current slot cannot be determined because the slot stack for this task is empty.
This may happen if you try to create UI from a background task.
To fix this, enter the target slot explicitly using `with container_element:`.

```

**Reply by rodja** [reply]
See [the docs](https://nicegui.io/documentation/run_javascript#run_javascript:~:text=If%20the%20function%20is%20awaited%2C%20the%20result%20of%20the%20JavaScript%20code%20is%20returned.%20Otherwise%2C%20the%20JavaScript%20code%20is%20executed%20without%20waiting%20for%20a%20response.) for when to await/not await `run_javascript`. But I don't understand what the `raise RuntimeError...` has to do with it.

**Reply by stepheweffie** [reply]
That's what happens if I don't await it. Server error shutdown.

**Reply by stepheweffie** [reply]
<img width="1440" alt="Screen Shot 2023-12-30 at 6 58 55 PM" src="https://github.com/zauberzeug/nicegui/assets/1074503/6db65775-d4b9-40b1-9f85-e1f97e361cb5">
I have helpful hints in PyCharm, I realize everyone else is using VS.

**Comment by rodja** [comment]
Can you please boil it down to a minimal executable example? That would allow us to reproduce the error.

**Reply by stepheweffie** [reply]
OMG, I don't need you to do that. Don't worry about me. I'm just here to help unless I really do need help then you'll know. I discuss for discussion not just to get help. Let me help.

**Comment by stepheweffie** [comment]
Runs fabulous with a helper return 

```
async def editor_js():
    editor = await ui.run_javascript(vue_js_script)
    return editor

with ui.expansion("WSIWYG Editor", icon='editor').classes('w-full ml-5'):
    ui.html(vue_html_content)
    await editor_js()
```

No error. Just relax. I'm working on a LaTEX renderer, my blog is using sqlalchemy not Tortoise and I have tons of tips on using this framework.
